### PR TITLE
fix: add test to prove redocVersion cannot be set

### PR DESCRIPTION
**Pull request by jarlah**
this can also serve as a solid foundation for more tests.

This PR only adds a fix to the redocVersion not being overridable. Im not even trying to fix config.ts because I don't think its possible without refactoring a lot.

also updated all packages and fixed a typescript issue in config.ts

### CODE REVIEW

**Comment by jarlah** [comment]
@amorimjj would be jolly nice if you took a look at this, even if you're as busy everyone else.

File: src/config/config.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ export abstract class Config<T> {
 
     public constructor(options: T) {
         this.options = options;
-        for ( const key of Object.keys(options) as Array<keyof Config<T>> ) {
+        for ( const key of Object.keys(options as any) as Array<keyof Config<T>> ) {
```
Comment (jarlah) [inline src/config/config.ts:6]: this was the typescript issue that I "fixed" by casting options to any

File: src/config/redoc-try-it-out-config.ts [hunk 2]
```diff
@@ -21,6 +21,8 @@ export class RedocTryItOutConfig extends Config<RedocTryItOutOptions> implements
 
     public constructor(docUrl: string, cfg:RedocTryItOutOptions, element?: HTMLElement) {
         super(cfg);
+        // uncomment to "fix" test
+        // this.redocVersion = cfg.redocVersion || '2.0.0-rc.56';
```
Comment (jarlah) [inline src/config/redoc-try-it-out-config.ts:25]: if this uncommented, the test will pass

**Comment by jarlah** [comment]
the this reference in javascript is a major bug nest

**Comment by jarlah** [comment]
after running the test, the config object looks like this

```typescript
      RedocTryItOutConfig {
        options: { redocVersion: 'foo' },
        _containerId: 'redoc-container',
        _operationBoxSelector: '[data-section-id]',
        tryItOutEnabled: true,
        tryItBoxContainerId: 'try-out-wrapper',
        redocVersion: '2.0.0-rc.56',
        selectedOperationClass: 'try',
        dependenciesVersions: { jquery: '3.6.0', jqueryScrollTo: '2.1.2' }
      }
```

**Comment by jarlah** [comment]
for anyone that is interested in using this try it out feature and override redocVersion, you can use this cdn link 

https://cdn.jsdelivr.net/gh/jarlah/redoc-try-it-out@b7723f31d9e1ab8488334fc18eea635022899dfa/dist/try-it-out.min.js

my forks is only containing this PR plus the minified try it out file in dist folder.

**Comment by jarlah** [comment]
if the original package is discontinued, im more than happy to extend the fork to fix all the other issues as well. I have a couple of active open source projects in my belt already.

**Comment by homestar9** [comment]
This is awesome. Thank you for doing that.  I don't think this repo is maintained anymore, and I would very much like to see an updated fork.  I'm sure many others would as well. :)

**Comment by jarlah** [comment]
moved my fork here btw 

https://github.com/RedocCommunity/redoc-try-it-out

Latest version of js now being accessible at

https://cdn.jsdelivr.net/gh/RedocCommunity/redoc-try-it-out@main/dist/try-it-out.min.js

Dont think ill bother creating an npm package, unless someone sees the need for it.

If you want to contribute you are free to do so, @homestar9, or anyone else for that matter.

**Comment by jarlah** [comment]
hopefully "RedocCommunity" is not breaching any naming rules or trademarks

**Comment by jarlah** [comment]
@homestar9 would you like to help contributing to the new organisation as an owner ? then we are at least two people that can drive it forward. Since its an org its not tied to a username too, which I think "feels better"

**Comment by homestar9** [comment]
@jarlah, thank you. Yeah, that sounds good to me.  Typescript isn't a language I am an expert in by any means, but I am happy to contribute in any way(s) I can.

**Comment by jarlah** [comment]
@homestar9 cool cool. I feel there is too many repos out there that fall into the trap of only having 1 owner or 1 maintainer. Lets try to avoid that this time around ;) Its usable already, even if the swagger UIs try out table is awkwardly squished together. It's not meant to be in a sidebar, it's meant to use more space. Maybe it could use modal for the try out feature. Anyway lets continue in the new repo :=)

**Comment by homestar9** [comment]
Agreed!  Sounds like a plan. :)

**Pull request closed without merging**